realm/realm-library/src/androidTest/java/io/realm/LinkingObjectsManagedTests.java

```diff
@@ -0,0 +1,741 @@
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .isNotNull("objectParents.listParents")
+            .findAll();
+        assertEquals(2, result.size());
+        assertTrue(result.contains(gen2A));
+        assertTrue(result.contains(gen2B));
+    }
+
+    // Query on a field descriptor that has a backlink in the middle
+    // The test objects are:
+    //             gen1
+    //             / \
+    //         gen2A gen2B
+    //           \\   //
+    //            gen3
+    //  /  = object ref
+    //  // = list ref
+    @Test
+    @Ignore
+    public void query_backlinkInMiddle() {
+        realm.beginTransaction();
+        AllJavaTypes gen1 = realm.createObject(AllJavaTypes.class, 10);
+
+        AllJavaTypes gen2A = realm.createObject(AllJavaTypes.class, 1);
+        gen2A.setFieldObject(gen1);
+
+        AllJavaTypes gen2B = realm.createObject(AllJavaTypes.class, 2);
+        gen2B.setFieldObject(gen1);
+
+        AllJavaTypes gen3 = realm.createObject(AllJavaTypes.class, 3);
+        RealmList<AllJavaTypes> parents = gen3.getFieldList();
+        parents.add(gen2A);
+        parents.add(gen2B);
+
+        realm.commitTransaction();
+
+        RealmResults<AllJavaTypes> result = realm.where(AllJavaTypes.class)
+            .lessThan("objectParents.listParents.fieldId", 4)
+            .findAll();
+        assertEquals(2, result.size());
+    }
+
+    // Based on a quick conversation with Christian Melchior and Mark Rowe,
+    // it appears that notifications are enqueued, briefly, on a non-Java
+    // thread.  That makes their delivery onto the looper thread unpredictable.
+    // Fortunately, it appears that beginning a transaction forces the delivery of
+    // any outstanding notifications.
+    // The closure passed to this method will be run *after* the body of the test
+    // completes, and *after* the notifications have been delivered.  Because the
+    // test method has been popped off the stack any objects referenced only from
+    // the stack are subject to GC.  To hang on to them, until the test completes,
+    // just pass them to the final vararg to this method.
+    // @zaki50 has some evidence that notifications are delivered on a commit.
+    // If that is the case, we may be able to eliminate the ugly begin-commit
+    // that is the prologue to this method.
+    private void verifyPostConditions(final Realm realm, final PostConditions test, final Object... refs) {
+        realm.beginTransaction();
+        realm.commitTransaction();
+
+        // Runnable is guaranteed to be enqueued on the Looper queue, after the notifications
```

Comment:
Possibly true.  There is no proof.  There *is* proof that this works.  Until there is proof that we do not need the extra transaction, I'm leaving it.  This is a test!